pkg/controller/virtualnode.go

```diff
@@ -279,6 +274,17 @@ func vnodeNeedsUpdate(desired *appmeshv1beta1.VirtualNode, target *aws.VirtualNo
 		}
 	}
 
+	if desired.Spec.BackendDefaults != nil {
```

Comment:
if we fix `target.BackendDefaults() to return 'nil'` when `target.Data.Spec.BackendDefaults == nil`, this can be simplified to just be `!reflect.DeepEqual(desired.Spec.BackendDefaults, target.BackendDefaults())`